util/gnmi.go

```diff
@@ -42,6 +42,29 @@ func PathMatchesPrefix(path *gpb.Path, prefix []string) bool {
 	return true
 }
 
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// PathPartiallyMatchesPrefix reports whether the path partially or wholly
+// matches the prefix.
+//
+// e.g. a/b partially matches a/b/c or a/b, but doesn't match a/c.
+func PathPartiallyMatchesPrefix(path *gpb.Path, prefix []string) bool {
+	for len(prefix) != 0 && prefix[len(prefix)-1] == "" {
```

Comment:
when are there empty strings in path?

Reply:
This is just to guard against a trailing slash when splitting. I wouldn't have written this myself -- it is kept in uniformity with the existing `PathMatchesPrefix`

Reply:
This is just to guard against a trailing slash when splitting. I wouldn't have written this myself -- it is kept in uniformity with the existing `PathMatchesPrefix`